### PR TITLE
feat: Update agent and team multi-select with new multi-select dropdown

### LINKED ISSUE #2476: Update agent/team multiselect styles

### PR DESCRIPTION
**Pull request by iamsivin**
https://user-images.githubusercontent.com/64252451/125617017-a0f58847-9fe1-4e55-b6b3-7a3cf0547547.mov

Fixes #2476

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 1]
```diff
@@ -137,7 +133,7 @@ import ContactDetailsItem from './ContactDetailsItem.vue';
 import ContactInfo from './contact/ContactInfo';
 import ConversationLabels from './labels/LabelBox.vue';
 import ContactCustomAttributes from './ContactCustomAttributes';
-import AvailabilityStatusBadge from 'dashboard/components/widgets/conversation/AvailabilityStatusBadge.vue';
+import multiselectDropdown from 'shared/components/ui/MutiselectDropdown.vue';
```
Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:136]: Component name always should be in [**PascalCase**](https://vuejs.org/v2/style-guide/#Component-name-casing-in-JS-JSX-strongly-recommended)

Comment (muhsin-k) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:136]: import MultiselectDropdown from 'shared/components/ui/MutiselectDropdown.vue';

File: app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue [hunk 2]
```diff
@@ -331,6 +326,23 @@ export default {
       };
       this.assignedAgent = selfAssign;
     },
+    onClickAssignAgent(selectedItem) {
+      if (this.assignedAgent && this.assignedAgent.id === selectedItem.id) {
+        this.assignedAgent = null;
+      } else {
+        this.assignedAgent = selectedItem;
+      }
+      return this.assignedAgent;
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/conversation/ContactPanel.vue:335]: Return is not needed this and below method

File: app/javascript/shared/components/ui/MultiselectDropdownItems.vue [hunk 3]
```diff
@@ -102,17 +108,42 @@ export default {
...
       this.$emit('click', option);
     },
-
     focusInput() {
       this.$refs.searchbar.focus();
     },
+    activeDOMElement() {
+      return this.$refs.dropdownItem.querySelectorAll(
+        'ul > li > button.dropdown-item'
+      )[this.arrowIndex];
```
Comment (nithindavid) [inline app/javascript/shared/components/ui/MultiselectDropdownItems.vue:126]: This is creating a conflict when navigating with up/down keys after tab press. We need to keep them in sync.

**Review by nithindavid — CHANGES_REQUESTED** [review]

File: app/javascript/shared/components/ui/MultiselectDropdownItems.vue [hunk 4]
```diff
@@ -102,17 +107,52 @@ export default {
...
+    onArrowDown() {
+      const array = this.$refs.multiselectDropdown.querySelectorAll(
+        '.dropdown .multiselect-dropdown--item'
+      );
+      const focusedElement = this.$refs.multiselectDropdown.querySelector(
+        '.dropdown .multiselect-dropdown--item:focus'
+      );
+      const lastElement = array[array.length - 1];
+      const activeElementIndex = [...array].indexOf(focusedElement);
+      const lastElementIndex = [...array].indexOf(lastElement);
```
Comment (nithindavid) [inline app/javascript/shared/components/ui/MultiselectDropdownItems.vue:148]:
```suggestion
      const lastElementIndex = array.length -1 
```

File: app/javascript/shared/components/ui/MultiselectDropdownItems.vue [hunk 5]
```diff
@@ -102,17 +107,52 @@ export default {
...
+
+      if (activeElementIndex >= 1) {
+        array[activeElementIndex - 1].focus();
+      } else {
+        array[lastElementIndex].focus();
+      }
+    },
+    onArrowDown() {
+      const array = this.$refs.multiselectDropdown.querySelectorAll(
```
Comment (nithindavid) [inline app/javascript/shared/components/ui/MultiselectDropdownItems.vue:140]: Lets come up with a better name for `array`

**Review by nithindavid — CHANGES_REQUESTED** [review]
Please see the comments.

**Review by nithindavid — APPROVED** [review]

**Pull request merged**